code/workspaces/auth-service/src/userManagement/authZeroUserManagement.js

```diff
@@ -18,30 +20,57 @@ const authKeyMapping = {
   user_id: 'userId',
 };
 
-export function asyncGetUsers() {
-  return requestAccessToken(accessTokenRequest())
-    .then(bearer => axios.get(`${authZeroManagementApi}/users`, {
-      params: {
-        fields: 'name,user_id',
-        per_page: '100',
-      },
-      headers: {
-        Authorization: `Bearer ${bearer}`,
-      },
-    }).then(extractUsers)
-      .then(processUsers)
-      .catch(() => {
-        throw new Error('Unable to retrieve users from User Management Service.');
-      }))
-    .catch((err) => {
+export async function asyncGetUsers() {
+  try {
+    try {
+      let page = 0;
+      let users = [];
+      let extractedUsers = [];
+      let activeUsers = [];
+      let allUsers = [];
+
+      const tokenRequest = await accessTokenRequest();
+      const bearer = await requestAccessToken(tokenRequest);
+
+      // Below approach is required as the auth0 API results are paginated, with a maximum
+      // overall limit, as well as a maximum per_page limit. This approach (non-async) is
+      // taken to prevent excessive API calls which impact performance.
+      /* eslint-disable no-await-in-loop */
+      do {
+        users = await fetchUsers(bearer, page);
+        extractedUsers = await extractUsers(users);
+        activeUsers = extractedUsers.filter(user => user.name);
+        allUsers = allUsers.concat(activeUsers);
```

Comment:
How come the filter has been refactored into the loop rather than being left in the `extractUsers` function? I can't see that `activeUsers` is being used for anything other than as an argument in the following line.

Reply:
That makes sense. I can't think of a better way of doing it...
Might be worth adding the comment about why you can get the active users from filtering on `user.name` like there is in the old `extractUsers` function so that that knowledge is retained.